api/client/attach.go

```diff
@@ -30,25 +30,29 @@ func (cli *DockerCli) CmdAttach(args ...string) error {
 		return err
 	}
 
-	env := engine.Env{}
-	if err := env.Decode(stream); err != nil {
+	var c struct {
```

Comment:
It's the best solution for now, but maybe the refactoring problem comes from the fact that runconfig is a mess too. So maybe we could add a little TODO with a question like: `Should we reconsider having a type in api/types with a refactored runconfig?`

@LK4D4 wdyt?